pkg/util/provider/driver/driver.go

```diff
@@ -64,6 +74,32 @@ type CreateMachineResponse struct {
 	LastKnownState string
 }
 
+// InitializeMachineRequest encapsulates params for the VM Initialization operation in the Driver facade.
+type InitializeMachineRequest struct {
+	// Machine object representing VM that must be initialized
+	Machine *v1alpha1.Machine
+
+	// MachineClass backing the machine object
+	MachineClass *v1alpha1.MachineClass
+
+	//  Secret backing the machineClass object
+	Secret *corev1.Secret
+}
+
+// InitializeMachineResponse is the response for VM instance initialization (Driver.InitializeMachine).
+type InitializeMachineResponse struct {
+	// ProviderID is the unique identification of the VM at the cloud provider.
+	// ProviderID typically matches with the node.Spec.ProviderID on the node object.
+	// Eg: gce://project-name/region/vm-ID
+	ProviderID string
+
+	// NodeName is the name of the node-object registered to kubernetes.
+	NodeName string
+
+	// LastKnownState represents the last state of the VM instance after initialization operation.
+	LastKnownState string
```

Comment:
I looked at all the providers to see how `LastKnownState` was being used. For GCP, this is populated with a readable message field. See https://github.com/gardener/machine-controller-manager-provider-gcp/blob/8e6f154abbace7c6fb4a85051191de8211739677/pkg/gcp/machine_controller.go#L91


```go
response := &driver.CreateMachineResponse{
		ProviderID:     providerID,
		NodeName:       req.Machine.Name,
		LastKnownState: fmt.Sprintf("Created %s", providerID),
	}
```
Can discuss with you/rishabh if you are un-happy with this. Not religious about this and can remove this if needed.

Reply:
Removed this, but according to `docs/development/machine_error_codes.md`

LastKnownState is an OPTIONAL field that can store details of the last known state of the VM. It can be used by future operation calls to determine current infrastucture state. This state is saved on the machine object.

What the spec says and what the drivers implement are 2 different things.

Reply:
OK, confirmed we can ignore this now.